### PR TITLE
Combine OrtValue and MLValue into one type

### PR DESCRIPTION
**Pull request by snnn**
Will do the cleanup after this PR.

### CODE REVIEW

File: onnxruntime/core/framework/tensor_type_and_shape.cc [hunk 1]
```diff
@@ -126,15 +126,15 @@ OrtStatus* GetTensorShapeAndType(const onnxruntime::TensorShape* shape,
 ORT_API_STATUS_IMPL(OrtGetTensorShapeAndType, _In_ const OrtValue* value,
                     _Out_ OrtTensorTypeAndShapeInfo** out) {
   API_IMPL_BEGIN
-  auto v = reinterpret_cast<const ::onnxruntime::MLValue*>(value);
+  auto v = (value);
```
Comment (weixingzhang) [inline onnxruntime/core/framework/tensor_type_and_shape.cc]: The bracket can be removed.

File: onnxruntime/core/framework/tensor_type_and_shape.cc [hunk 2]
```diff
@@ -126,15 +126,15 @@ OrtStatus* GetTensorShapeAndType(const onnxruntime::TensorShape* shape,
...
   const onnxruntime::Tensor& tensor = v->Get<onnxruntime::Tensor>();
   return GetTensorShapeAndType(&tensor.Shape(), tensor.DataType(), out);
   API_IMPL_END
 }
 
 ORT_API(enum ONNXType, OrtGetValueType, _In_ const OrtValue* value) {
   try {
-    auto v = reinterpret_cast<const ::onnxruntime::MLValue*>(value);
+    auto v = (value);
```
Comment (weixingzhang) [inline onnxruntime/core/framework/tensor_type_and_shape.cc]: same as above.

File: onnxruntime/core/framework/tensor_type_and_shape.cc [hunk 3]
```diff
@@ -156,7 +156,7 @@ ORT_API(enum ONNXType, OrtGetValueType, _In_ const OrtValue* value) {
  * \return The returned value should be freed by OrtReleaseTypeInfo after use
  */
 ORT_API_STATUS_IMPL(OrtGetTypeInfo, _In_ const OrtValue* value, struct OrtTypeInfo** out) {
-  auto v = reinterpret_cast<const ::onnxruntime::MLValue*>(value);
+  auto v = (value);
```
Comment (weixingzhang) [inline onnxruntime/core/framework/tensor_type_and_shape.cc]: same as above.

**Comment by weixingzhang** [comment]
To match new name of type, should the functions or parameters names be changed accordingly? Or you plan to do it in a separate PR?

**Comment by snnn** [comment]
Hi @weixingzhang, I know your concern. There are a lot of clean up work need be done after this PR.  In order to make every change smaller and code review easier, I'll split them into multiple PRs.

File: include/onnxruntime/core/framework/ml_value.h [hunk 4]
```diff
@@ -10,20 +10,19 @@
 #include "core/framework/data_types.h"
 #include "core/framework/tensor.h"
 
-namespace onnxruntime {
 /**
    Represents both tensors and non-tensors.
 */
-class MLValue {
+struct OrtValue {
  public:
```
Comment (skottmckay) [inline include/onnxruntime/core/framework/ml_value.h]: Should we change this filename in this PR?

**Pull request merged**